src/modules/rooms/rooms.graphql

```diff
@@ -28,3 +28,18 @@ type Room {
   doctor: Doctor
   createdAt: String!
 }
+
+# ------------------------------ Response types ------------------------------ #
+type InfoForGuest {
+  id: ID!
+  name: String!
+}
+
+type InfoForAdmin {
+  id: ID!
+  name: String!
+  doctorId: String!
+}
+
+# ---------------------------------- Unions ---------------------------------- #
+union RestrictedRoom = InfoForAdmin | InfoForGuest
```

Comment:
I guess it can be done by defining a nullable field (doctorId) and then creating a resolveField handler where you will return null depending on the user's role.

Reply:
Also, you can use graphql directives here

Reply:
I need help, to understand how to implement directives and call context inside of them